src/Bridge/Doctrine/Orm/Extension/EagerLoadingExtension.php

```diff
@@ -174,7 +187,16 @@ private function joinRelations(QueryBuilder $queryBuilder, QueryNameGeneratorInt
                 continue;
             }
 
-            $this->joinRelations($queryBuilder, $queryNameGenerator, $mapping['targetEntity'], $forceEager, $fetchPartial, $associationAlias, $propertyMetadataOptions, $method === 'leftJoin', $joinCount);
+            if (null !== $attributesMetadata && isset($attributesMetadata[$association])) {
+                $maxDepth = $attributesMetadata[$association]->getMaxDepth();
+
+                // The current depth is the lowest max depth available in the ancestor tree.
+                if (null !== $maxDepth && ($maxDepth < $currentDepth || null === $currentDepth)) {
```

Comment:
I tried your suggestion but it throws the max join error  as null < $currentDepth satisfies the condition, which we don't want.  I changed the order a bit to 
(null !== $maxDepth && (null === $currentDepth || $maxDepth < $currentDepth))